lib/lib.ml

```diff
@@ -2,13 +2,258 @@
+      let hi = lo + try Option.get (substr "\r\n" hstr#.(lo,0)) with
+          Invalid_argument _ -> assert false (* should never happen if libpst does its job *)
+      in
+      let old_name =    (* offsets are for escaped quotation marks and/or a ';',
+                           which is only required if more params follow the filename param.*)
+        if mem ';' hstr#.(lo, hi)
+        then hstr#.(lo + 10, hi - 2)
+        else hstr#.(lo + 10, hi - 1)
+      in
+      let new_name = (remove_extension old_name)
+                     ^ ".CONVERTED"
+                     ^ timestamp ()
+                     ^ (extension old_name)
+                     ^ ext
+      in replace old_name new_name hstr
+    | _ -> assert false
+
+  (* TESTS *)
+
+  (* TODOs
+   *
+   * - [x] have pdf_convert_test update the Content-Type header
+   * - [~] put the line breaks back in the semicolon-separated header field
+   *   parameters (i.e. filename=etc) in the output
+   * - [x] see if the result will open in Apple Mail
+   *
+   * to make the output into an MBOX, put this at the top:
+   *
+From root@gringotts.lib.uchicago.edu Fri Jan 21 11:48:27 2022
+   *)
+
+  let docx_convert_test fname =
+    let header_func hstring =
+      let hs = String.lowercase_ascii hstring in
+      match Strings.(
+          (substr "content-disposition: attachment;" hs,
+           substr "content-type: application/vnd.openxmlformats-officedocument.wordprocessingml.document" hs)
+        )
+      with
+        (Some _, Some _) -> update_mimetype
+                              "application/vnd.openxmlformats-officedocument.wordprocessingml.document" (* note: should we make this optional? how much could we infer from config etc *)
+                              "application/pdf"
+                              (update_filename ~ext:".pdf" hstring)
+      | _ -> hstring (* noop when not a pdf and attachment *)
+    in
+    let body_func bstr =
+      let open Prelude.Unix.Proc in
+      let tmpname = fname ^ "_extracted_tmp.docx" in
+      (writefile ~fn:(tmpname) bstr; read ["pandoc"; "--from=docx"; "--to=pdf"; "--pdf-engine=xelatex"; tmpname])
+    in
+    let tree = parse (readfile fname) in
+    let converted_tree = amap header_func body_func tree
+    in writefile ~fn:(fname ^ "_docxmas_saved") (to_string converted_tree)
+
+
+  let upcase_header_and_delete_body fname =
+    let f = String.uppercase_ascii in
+    let g = fun _ -> "" in
+    let tree = parse (Prelude.readfile fname) in
+    Prelude.writefile ~fn:(fname ^ "_upcased_and_deleted") (tree |> (amap f g) |> to_string)
+
+  let omit_gore_y_details fname =
+    let f = Fun.const "Content-Type: application/json\r\n\r\n" in
+    (* let f = fun _ -> "From: Bill Clinton <president@whitehouse.gov>\r\n\r\n" in *)
+    let g = Fun.const "" in
+    let tree = parse (Prelude.readfile fname) in
+    Prelude.writefile ~fn:(fname ^ "_contented") (tree |> (amap f g) |> to_string)
+
+  let extra_spaces_in_header fname =
```

Comment:
I added a small test for adding extra spaces. Again, not sure if this transformation should be allowed based on the MIME spec, but it might be looking into.